### DISCUSSION TITLE
Can the selected tab be programmatically set?

### DISCUSSION THREAD

**Discussion by brandonpittman** [johnlindquist/kit#561] [Q&A] [answered]
I have a use-case where I have "New" and "Reset" tabs. On the "Reset" tab, there are options for "Reset" and "Cancel". When selecting one, I want to move the selected tab back over to "New". Is this possible right now?

**Comment by johnlindquist** [comment, marked as answer]
Possible? Yes. Does it have an api, not really 😅

`send` communicates with the app to update the UI. A lot of the internals use this.
`onTabs` is an array with name/fn

```js
let setTab = tabName => {
  let i = global.onTabs.findIndex(
    ({ name }) => name === tabName
  )
  global.send("SET_TAB_INDEX", { tabIndex: i })
  global.onTabs[i].fn()
}

let newTab = async () => {
  await arg("New", ["a", "b", "c"])
}

let resetTab = async () => {
  let value = await arg("Reset", ["Reset", "Cancel"])
  if (value === "Reset") {
    setTab("New")
  }
}

onTab("New", newTab)
onTab("Reset", resetTab)
```

`setTab` probably belongs in the API... 🤔

**Reply by brandonpittman** [reply]
@johnlindquist Lovely! Looking forward to seeing it as part of the API. 🚀 

https://user-images.githubusercontent.com/967145/140835812-afffdb0f-ba2c-4aa8-a704-bc881457acd2.mp4

**Reply by johnlindquist** [reply]
Ah, I see what you're doing.

Here's a different way of thinking about the same problem space:

A `while(true)` wrapped around `editor` allows you to submit as many times as you want.

With `editor` <kbd>cmd+s</kbd> will submit. If you type "clear!" (or some other word of your choosing) at the end of the file, it will erase the contents.

```js
let filePath = tmpPath("thoughts.md") //Choose your own path! Make your own destiny!

await ensureFile(filePath)

let contents = await readFile(filePath, "utf-8")

while (true) {
  contents = await editor(contents)

  if (contents.endsWith("clear!")) {
    contents = ""
  }

  await writeFile(filePath, contents)
}
```

Of course you can always just <kbd>cmd+a</kbd> select all, delete, and save. So you have plenty of options.


https://user-images.githubusercontent.com/36073/140837296-cac19fcf-8903-4519-9223-5f9fca2e8879.mp4

Never underestimate the power of a simple text file :)

**Reply by brandonpittman** [reply]
Not sure why, but this script isn't working at all for me. The text file never updates. 🤔

**Reply by johnlindquist** [reply]
Are you hitting cmd+s to submit the editor contents?

**Reply by brandonpittman** [reply]
Ha! No, I totally blanked and skipped that sentence. Sorry. 🤦🏻‍♂️
